### PR TITLE
Function for pooling the prediction of a lm model

### PR DESCRIPTION
**Pull request by Fdvanleeuwen**
The function allows a user to pool predictions for an lm model, when there are multiple prediction models, multiple imputed test cases, or both. The output are the mean predictions, with a prediction interval that is constructed conform Rubin's rules.

### CODE REVIEW

**Comment by Fdvanleeuwen** [comment]
This solves #82 for lm models. Here a repress to show the functionality:

# Dataframe with missings in X
dat <- mice::nhanes

# Split the data in training and test set
train = dat[1:20,]
test = dat[21:25,]

# Impute missing values in train, and make an imputation model to be used for test
imp_train = mice::mice(train, m = 5, maxit = 5, task = "train")
#> 
#>  iter imp variable
#>   1   1  age  bmi  hyp  chl
#>   1   2  age  bmi  hyp  chl
#>   1   3  age  bmi  hyp  chl
#>   1   4  age  bmi  hyp  chl
#>   1   5  age  bmi  hyp  chl
#>   2   1  age  bmi  hyp  chl
#>   2   2  age  bmi  hyp  chl
#>   2   3  age  bmi  hyp  chl
#>   2   4  age  bmi  hyp  chl
#>   2   5  age  bmi  hyp  chl
#>   3   1  age  bmi  hyp  chl
#>   3   2  age  bmi  hyp  chl
#>   3   3  age  bmi  hyp  chl
#>   3   4  age  bmi  hyp  chl
#>   3   5  age  bmi  hyp  chl
#>   4   1  age  bmi  hyp  chl
#>   4   2  age  bmi  hyp  chl
#>   4   3  age  bmi  hyp  chl
#>   4   4  age  bmi  hyp  chl
#>   4   5  age  bmi  hyp  chl
#>   5   1  age  bmi  hyp  chl
#>   5   2  age  bmi  hyp  chl
#>   5   3  age  bmi  hyp  chl
#>   5   4  age  bmi  hyp  chl
#>   5   5  age  bmi  hyp  chl

# Fit the prediction models, based on the imputed training data
fits = with(imp_train, lm(age ~ bmi + hyp + chl))

# To ensure the data types match between train and test use mice::coerce
# Fit the imputation model on the test set
test_imp <- mice::mice(test, tasks = "fill", models = imp_train$models)
#> 
#>  iter imp variable
#>   1   1  age  bmi  hyp  chl
#>   1   2  age  bmi  hyp  chl
#>   1   3  age  bmi  hyp  chl
#>   1   4  age  bmi  hyp  chl
#>   1   5  age  bmi  hyp  chl
#>   2   1  age  bmi  hyp  chl
#>   2   2  age  bmi  hyp  chl
#>   2   3  age  bmi  hyp  chl
#>   2   4  age  bmi  hyp  chl
#>   2   5  age  bmi  hyp  chl
#>   3   1  age  bmi  hyp  chl
#>   3   2  age  bmi  hyp  chl
#>   3   3  age  bmi  hyp  chl
#>   3   4  age  bmi  hyp  chl
#>   3   5  age  bmi  hyp  chl
#>   4   1  age  bmi  hyp  chl
#>   4   2  age  bmi  hyp  chl
#>   4   3  age  bmi  hyp  chl
#>   4   4  age  bmi  hyp  chl
#>   4   5  age  bmi  hyp  chl
#>   5   1  age  bmi  hyp  chl
#>   5   2  age  bmi  hyp  chl
#>   5   3  age  bmi  hyp  chl
#>   5   4  age  bmi  hyp  chl
#>   5   5  age  bmi  hyp  chl

# Retrieve the imputed datasets
test_imp = mice::complete(test_imp, "all")

# Pool the predictions for the test set
head(pool.preds.lm(fits, test_imp))
#>         fit        lwr      upr
#> 1 0.9197579 -1.5017262 3.341242
#> 2 1.1949251 -1.2256161 3.615466
#> 3 0.8256843 -1.5001996 3.151568
#> 4 1.9129405 -0.5779726 4.403854
#> 5 1.3711653 -0.8414070 3.583738
